.travis.yml

```diff
@@ -17,30 +17,59 @@ env:
-    - PANDAS_VERSION=0.18.1
-    - SCIPY_VERSION=0.17.1
+  matrix:
+    - PANDAS_18=1
+    - PANDAS_22=1
 matrix:
+  exclude:
+    - python: 2.7
+      env: PANDAS_22=1
   include:
-    # - python: 3.5
-    #   env: PANDAS_VERSION=0.19.2
-    - python: 3.5
-      env: NUMPY_VERSION=1.14.1 PANDAS_VERSION=0.22.0 SCIPY_VERSION=1.0.0 PANDAS_DATAREADER_VERSION=0.4.0 DASK_VERSION=0.17.1
+    # Workaround Travis OSX not natively supporting Python.
+    - os: osx
+      language: generic
+      env: TRAVIS_PYTHON_VERSION=2.7 PANDAS_18=1
+    - os: osx
+      language: generic
+      env: TRAVIS_PYTHON_VERSION=3.5 PANDAS_18=1
+    - os: osx
+      language: generic
+      env: TRAVIS_PYTHON_VERSION=3.5 PANDAS_22=1
+
 cache:
   directories:
     - $HOME/.cache/.pip/
 
 before_install:
-  - if [ ${CONDA_ROOT_PYTHON_VERSION:0:1} == "2" ]; then wget https://repo.continuum.io/miniconda/Miniconda2-4.3.30-Linux-x86_64.sh -O miniconda.sh; else wget https://repo.continuum.io/miniconda/Miniconda3-4.3.30-Linux-x86_64.sh -O miniconda.sh; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then MINICONDA_OS=MacOSX; else MINICONDA_OS=Linux; fi
+  - wget "https://repo.continuum.io/miniconda/Miniconda${CONDA_ROOT_PYTHON_VERSION:0:1}-4.3.30-$MINICONDA_OS-x86_64.sh" -O miniconda.sh
   - chmod +x miniconda.sh
   - ./miniconda.sh -b -p $HOME/miniconda
   - export PATH="$HOME/miniconda/bin:$PATH"
-  - sed -i "s/numpy==.*/numpy==$NUMPY_VERSION/" etc/requirements.txt
-  - sed -i "s/pandas==.*/pandas==$PANDAS_VERSION/" etc/requirements.txt
-  - sed -i "s/scipy==.*/scipy==$SCIPY_VERSION/" etc/requirements.txt
-  - if [ -n "$PANDAS_DATAREADER_VERSION" ]; then sed -i "s/pandas-datareader==.*/pandas-datareader==$PANDAS_DATAREADER_VERSION/" etc/requirements.txt; fi
-  - if [ -n "$DASK_VERSION" ]; then sed -i "s/dask\[dataframe\]==.*/dask\[dataframe\]==$DASK_VERSION/" etc/requirements_blaze.txt; fi
+  - |
+    if [ "$PANDAS_18" ]; then
+      NUMPY_VERSION=1.11.1 PANDAS_VERSION=0.18.1 SCIPY_VERSION=0.17.1
+    else
+      NUMPY_VERSION=1.14.1 PANDAS_VERSION=0.22.0 SCIPY_VERSION=1.0.0 PANDAS_DATAREADER_VERSION=0.4.0 DASK_VERSION=0.17.1
+    fi
+  - |
+    if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+      # On OSX, sed refuses to edit in place, so give it an empty extension for the rename.
+      function sed_inplace() {
+        sed -i '' "$@"
+      }
+    else
+      function sed_inplace() {
+        sed -i "$@"
+      }
+    fi
+  - sed_inplace "s/numpy==.*/numpy==$NUMPY_VERSION/" etc/requirements.txt
+  - sed_inplace "s/pandas==.*/pandas==$PANDAS_VERSION/" etc/requirements.txt
+  - sed_inplace "s/scipy==.*/scipy==$SCIPY_VERSION/" etc/requirements.txt
+  - if [ -n "$PANDAS_DATAREADER_VERSION" ]; then sed_inplace "s/pandas-datareader==.*/pandas-datareader==$PANDAS_DATAREADER_VERSION/" etc/requirements.txt; fi
+  - if [ -n "$DASK_VERSION" ]; then sed_inplace "s/dask\[dataframe\]==.*/dask\[dataframe\]==$DASK_VERSION/" etc/requirements_blaze.txt; fi
   - cat etc/requirements.txt
+
```

Comment:
Same question here for moving this into its own file instead of writing a bunch of bash inline.